--- a/arch/arm64/boot/dts/rockchip/rk3568.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568.dtsi
@@ -233,7 +233,7 @@
 &cpu0_opp_table {
 	opp-1992000000 {
 		opp-hz = /bits/ 64 <1992000000>;
-		opp-microvolt = <1150000 1150000 1150000>;
+		opp-microvolt = <1250000>;
 	};
 };
 
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -134,39 +134,39 @@
 
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <900000 900000 1150000>;
+			opp-microvolt = <1000000 975000 1150000>;
 			clock-latency-ns = <40000>;
 		};
 
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <900000 900000 1150000>;
+			opp-microvolt = <1000000 975000 1150000>;
 		};
 
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <900000 900000 1150000>;
+			opp-microvolt = <1000000 975000 1150000>;
 			opp-suspend;
 		};
 
 		opp-1104000000 {
 			opp-hz = /bits/ 64 <1104000000>;
-			opp-microvolt = <900000 900000 1150000>;
+			opp-microvolt = <1000000 975000 1150000>;
 		};
 
 		opp-1416000000 {
 			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <900000 900000 1150000>;
+			opp-microvolt = <1000000 975000 1150000>;
 		};
 
 		opp-1608000000 {
 			opp-hz = /bits/ 64 <1608000000>;
-			opp-microvolt = <975000 975000 1150000>;
+			opp-microvolt = <1000000 1000000 1150000>;
 		};
 
 		opp-1800000000 {
 			opp-hz = /bits/ 64 <1800000000>;
-			opp-microvolt = <1050000 1050000 1150000>;
+			opp-microvolt = <1150000 1000000 1250000>;
 		};
 	};
 
--- a/drivers/clk/rockchip/clk-rk3568.c
+++ b/drivers/clk/rockchip/clk-rk3568.c
@@ -162,16 +162,17 @@ static struct rockchip_pll_rate_table rk
 }
 
 static struct rockchip_cpuclk_rate_table rk3568_cpuclk_rates[] __initdata = {
+	RK3568_CPUCLK_RATE(1992000000, 0, 1, 8, 8, 8, 8),
 	RK3568_CPUCLK_RATE(1800000000, 0, 1, 7, 7, 7, 7),
 	RK3568_CPUCLK_RATE(1704000000, 0, 1, 7, 7, 7, 7),
-	RK3568_CPUCLK_RATE(1608000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1584000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1560000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1536000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1512000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1488000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1464000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1440000000, 0, 1, 5, 5, 5, 5),
+	RK3568_CPUCLK_RATE(1608000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1584000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1560000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1536000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1512000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1488000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1464000000, 0, 1, 6, 6, 6, 6),
+	RK3568_CPUCLK_RATE(1440000000, 0, 1, 6, 6, 6, 6),
 	RK3568_CPUCLK_RATE(1416000000, 0, 1, 5, 5, 5, 5),
 	RK3568_CPUCLK_RATE(1392000000, 0, 1, 5, 5, 5, 5),
 	RK3568_CPUCLK_RATE(1368000000, 0, 1, 5, 5, 5, 5),
@@ -181,17 +182,17 @@ static struct rockchip_cpuclk_rate_table
 	RK3568_CPUCLK_RATE(1272000000, 0, 1, 5, 5, 5, 5),
 	RK3568_CPUCLK_RATE(1248000000, 0, 1, 5, 5, 5, 5),
 	RK3568_CPUCLK_RATE(1224000000, 0, 1, 5, 5, 5, 5),
-	RK3568_CPUCLK_RATE(1200000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(1104000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(1008000000, 0, 1, 3, 3, 3, 3),
+	RK3568_CPUCLK_RATE(1200000000, 0, 1, 4, 4, 4, 4),
+	RK3568_CPUCLK_RATE(1104000000, 0, 1, 4, 4, 4, 4),
+	RK3568_CPUCLK_RATE(1008000000, 0, 1, 4, 4, 4, 4),
 	RK3568_CPUCLK_RATE(912000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(816000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(696000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(600000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(408000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(312000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(216000000, 0, 1, 3, 3, 3, 3),
-	RK3568_CPUCLK_RATE(96000000, 0, 1, 3, 3, 3, 3),
+	RK3568_CPUCLK_RATE(816000000, 0, 1, 3, 3, 3, 2),
+	RK3568_CPUCLK_RATE(696000000, 0, 1, 2, 2, 2, 2),
+	RK3568_CPUCLK_RATE(600000000, 0, 1, 2, 2, 2, 2),
+	RK3568_CPUCLK_RATE(408000000, 0, 1, 1, 1, 1, 1),
+	RK3568_CPUCLK_RATE(312000000, 0, 1, 1, 1, 1, 1),
+	RK3568_CPUCLK_RATE(216000000, 0, 1, 1, 1, 1, 1),
+	RK3568_CPUCLK_RATE(96000000, 0, 1, 1, 1, 1, 1),
 };
 
 static const struct rockchip_cpuclk_reg_data rk3568_cpuclk_data = {
